File: 길민지/Week_12/P118666_성격유형검사하기.py
def solution(survey, choices):
    score = {
        'R' : 0,
        'T' : 0,
        'C' : 0,
        'F' : 0,
        'J' : 0,
        'M' : 0,
        'A' : 0,
        'N' : 0,
    }

    # 점수 계산
    for i in range(len(choices)):
        if(choices[i]<4):
            score[survey[i][0]] += 4-choices[i]
        else:
            score[survey[i][1]] += choices[i]-4

    # 유형 결정
    answer = ''
    answer+= 'R' if score['R'] >= score['T'] else 'T'
    answer+= 'C' if score['C'] >= score['F'] else 'F'
    answer+= 'J' if score['J'] >= score['M'] else 'M'
    answer+= 'A' if score['A'] >= score['N'] else 'N'

    return answer

File: 이우엽/Week_12/P118666_성격유형검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        int[] scores = {-1, 3, 2, 1, 0, 1, 2, 3};
        Character[] types = {'R', 'T', 'C', 'F', 'J', 'M', 'A', 'N'};
        int[] scoreByTypes = {0, 0, 0, 0, 0, 0, 0, 0};
        for(int i = 0; i < choices.length; i++) {
            char type1 = survey[i].charAt(0);
            char type2 = survey[i].charAt(1);
            int type1Idx = Arrays.asList(types).indexOf(type1);
            int type2Idx = Arrays.asList(types).indexOf(type2);
            int choice = choices[i];
            switch(choice) {
                case 1, 2, 3:
                    scoreByTypes[type1Idx] += scores[choice];
                    break;
                case 5, 6, 7:
                    scoreByTypes[type2Idx] += scores[choice];
                    break;
                default:
                    break;
            }

        }
        for(int score : scoreByTypes) {
            System.out.println("score: " + score);
        }
        for(int i = 0; i < scoreByTypes.length; i = i + 2) {
            answer += scoreByTypes[i] >= scoreByTypes[i+1] ? types[i] : types[i+1];
        }
        return answer;

    }
}